### ISSUE TITLE
Client not connecting to server

### ISSUE THREAD

**Issue by wjflier2** [goldbattle/esp8266_thermostat#1]
Great idea but having issues.

Error message 'HTTPClient::begin' declared with attribute error: obsolete API, use ::begin(WiFiClient, url)
Changed code to: 
// Declare object of class HTTPClient
    HTTPClient http;
    WiFiClient client;

    // Setup the destination and content type
    http.begin(client, "http://"+server+"/reading");
    http.addHeader("Content-Type", "application/x-www-form-urlencoded");
no error, server and client both run, but no connection.

**Comment by goldbattle** [comment]
I assume you are using the v1 version of this code?
I no longer use this so can't help debug anything.


Maybe the libraries have updated since I used them.
Here is what I have on my machine:

- DHTesp - 1.0.11
- ESPAsyncTCP - 1.2.0
- ESP Async WebServer - 1.2.0

It looks like you might need to define a WiFiClient now, see this link:
https://stackoverflow.com/a/67702894/7718197

**Comment by goldbattle** [comment]
Oh, I didn't fully read what you said, sorry about that. I see you fixed the problem.

Maybe you can check `Serial.println("http response: "+httpCode);` to see if there is any connection there.
Are you able to view each devices' web interface?

**Comment by wjflier2** [comment]
Yes both web interfaces show, but the server doesn't update. header displays "ERROR: have not got any sensor readings from device office.
ERROR: last reading was from 19 hrs, 11 min, 18 sec ago.
ERROR: the system will be turned off till we get a new reading.

**Comment by wjflier2** [comment]
"thermostat.local.lan" doesn't bring up anythin, I use the server ip

**Comment by wjflier2** [comment]
I don't get where these connect, all I've read use AP to connect.  What am I missing?

**Comment by goldbattle** [comment]
You are connecting all the esp to the same wifi network correct?
They shouldn't be in AP mode.

**Comment by wjflier2** [comment]
yes

**Comment by wjflier2** [comment]
both client and server post on their respective web pages but don't seem to talk to each other

**Comment by goldbattle** [comment]
Are you able to ping each client yourself? Otherwise, I am not sure.

**Comment by wjflier2** [comment]
Yes, they ping, but I don't see where the server request data from the client, or the client broadcast to the server

**Comment by wjflier2** [comment]
Not sure if this is an issue, but I noticed this part of the code doesn't work: 
// Connect to Wi-Fi with our static ip
    WiFi.config(staticIP, subnet, gateway, dns);
    WiFi.begin(ssid, password);  
    Serial.println("Connecting to WiFi");
    while (WiFi.status() != WL_CONNECTED) {
        delay(1000);
        Serial.println(".");

does by changing subnet to (255, 255, 0, 0)  but then the server web page doesn't load

**Comment by wjflier2** [comment]
looks like something with dns
here's what I'm getting now till I reset
current temp of 0.00 < 75
current humidity is 0.00 %
no need to turn on cooling
user has forced fan to be on.

**Comment by wjflier2** [comment]
Just wanted to say thanks for your responses.  I'll figure it out, good brain exercise.
